### PR TITLE
[BugFix] Fix some wrong usages of down_cast

### PR DESCRIPTION
**Pull request by sduzh**
- Replace down_cast with dynamic_cast.
- Add some tests for LoadChannel, only cover LakeTablet related cases.

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

File: be/src/runtime/load_channel.cpp [hunk 1]
```diff
@@ -275,6 +279,8 @@ std::shared_ptr<TabletsChannel> LoadChannel::get_tablets_channel(int64_t index_i
         auto local_tablets_channel = dynamic_cast<LocalTabletsChannel*>(it->second.get());
         if (local_tablets_channel) {
             local_tablets_channel->incr_num_ref_senders();
+        } else {
```
Comment (chaoyli) [inline be/src/runtime/load_channel.cpp:282]: What's the usage of this `else` branch?

Comment (sduzh) [reply be/src/runtime/load_channel.cpp:282]: Tell others that we didn't forget to handle the else branch.

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by imay — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**